pyciemss/integration_utils/observation.py

```diff
@@ -9,19 +9,63 @@
 
 
 def load_data(
-    path: str, data_mapping: Dict[str, str] = {}
+    path: Union[str, pd.DataFrame], data_mapping: Dict[str, str] = {}
 ) -> Tuple[torch.Tensor, Dict[str, torch.Tensor]]:
     """
-    Load data from a CSV file.
+    Load data from a CSV file, or directly from a DataFrame.
 
-    - path: path to the CSV file
+    - path: path to the CSV file, or DataFrame
     - data_mapping: A mapping from column names in the data file to state variable names in the model.
         - keys: str name of column in dataset
         - values: str name of state/observable in model
     - If not provided, we will assume that the column names in the data file match the state variable names.
     """
 
-    df = pd.read_csv(path)
+    def check_data(data_path: Union[str, pd.DataFrame]):
+        # This function checks a dataset for formatting errors, and returns a DataFrame
+
+        # Read the data
+        if isinstance(data_path, str):
+            # If data_path is a string, assume it's a file path and read as a csv
+            data_df = pd.read_csv(data_path)
+        elif isinstance(data_path, pd.DataFrame):
+            # If data_path is a DataFrame, use it directly
+            data_df = data_path
+        else:
+            # If data_path is neither a string nor a DataFrame, raise an error
+            raise ValueError("data_path must be either a file path or a DataFrame.")
+
+        # Check that the first column name is "Timestamp"
+        if data_df.columns[0] != "Timestamp":
+            raise ValueError(
+                "The first column must be named 'Timestamp' and contain the time corresponding to each row of data."
+            )
+
+        # Check that there are no NaN values or empty entries
```

Comment:
Is this a constraint that we want to impose?  It might be better if we could handle ragged data, yes?

Reply:
@djinnome It would be better, but I'm not sure where to go with that. How would that propagate to calibrate? I think it's probably best to throw an error message for now, and create a new issue to handle ragged data in the future.